src/content/docs/en/guides/deploy/github.mdx

```diff
@@ -130,40 +63,103 @@ export default defineConfig({
     :::
 
-2. (Optional) If you pass environment variables to your Astro project during local development or when previewing builds, you will need to define any public variables in the `deploy.yml` file in order for them to be processed when you deploy to GitHub Pages. (See the [GitHub documentation on setting secrets](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) for adding private environment variables.)
+2. In your Astro config file `astro.config.mjs`, set [`site`](/en/reference/configuration-reference/#site) to the GitHub URL of your deployed site.
 
-   ```yaml title="deploy.yml"
-   jobs:
-      build:
-        runs-on: ubuntu-latest
-        steps:
-          - name: Checkout your repository using git
-            uses: actions/checkout@v4
-          - name: Install, build, and upload your site
-            uses: withastro/action@v3
-            env:
-              # Use single quotation marks for the variable value
-              PUBLIC_EVM_WALLET_ADDRESS: '0x4bFc229A40d41698154336aFF864f61083E76659'
-   ```
+    The value for `site` must be one of the following:
 
+    - The following URL based on your username: `https://<username>.github.io`
+    - The random URL autogenerated for a [GitHub Organization's private page](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site): `https://<random-string>.pages.github.io/`
 
-3. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.
+    ```js title="astro.config.mjs" ins={4}
+    import { defineConfig } from 'astro/config'
 
-4. Choose **GitHub Actions** as the **Source** of your site.  
+    export default defineConfig({
+      site: 'https://astronaut.github.io',
+    })
+    ```
 
-5. Commit the new workflow file and push it to GitHub.  
-</Steps>
+3. In `astro.config.mjs`, configure a value for [`base`](/en/reference/configuration-reference/#base).
+
+      A value for `base` is required so that Astro will treat your repository name (e.g. `/my-repo`) as the root of your website.
+
+      The value for `base` should be your repository’s name starting with a forward slash, for example `/my-blog`. This is so that Astro understands your website's root is `/my-repo`, rather than the default `/`.
+
+      ```js title="astro.config.mjs" ins={4-5}
+      import { defineConfig } from 'astro/config'
+  
+      export default defineConfig({
+        site: 'https://astronaut.github.io',
+        base: '/my-repo',
+      })
+      ```
+
+      You do not need a value for`base` if:
+
+      - Your page is served from the root folder.
+      - Your repository is located at `https://github.com/<USERNAME>/<USERNAME>.github.io`.
+
```

Comment:
```suggestion
```
I think I removed the need for this! 
- don't need to say not for a custom domain because it's clear that this is about deploying to a github.io domain
- described the root case inline as "if your repo name exactly matches your username", relating it to the choice of base being the repo name